src/github.com/matrix-org/dendrite/clientapi/httputil/parse.go

```diff
@@ -13,26 +13,49 @@
 package httputil
 
 import (
+	"fmt"
 	"net/http"
 	"strconv"
 	"time"
+
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	"github.com/matrix-org/util"
 )
 
+// ParseIntParam takes a request and parses the query param to an int
+// returns 0 when not present and -1 when parsing failed as int
+func ParseIntParam(req *http.Request, param string) (int64, *util.JSONResponse) {
+	paramStr := req.URL.Query().Get(param)
+	if paramStr == "" {
+		return 0, &util.JSONResponse{
```

Comment:
This case is not an error.

I'm guessing you wanted to make `ParseIntParam` do strict check for ts parameter ?

Reply:
If I want to parse an int of nothing that is some kind of error. At least sth. I would want to return to the caller. I thought it would be a good approach to return 0 in that case.
So the calling function can distinguish from a "cannot parse" error.